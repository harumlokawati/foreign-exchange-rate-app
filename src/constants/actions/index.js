export const FAILURE = 'FAILURE';
export const LOAD_RATE = 'LOAD_RATE';
export const LOAD_RATE_SUCCESS = 'LOAD_RATE_SUCCESS';
export const SET_AMOUNT = 'SET_AMOUNT';
export const SET_BASE = 'SET_BASE'
export const ADD_SYMBOL = 'ADD_SYMBOL';
export const REMOVE_SYMBOL = 'REMOVE_SYMBOL';
