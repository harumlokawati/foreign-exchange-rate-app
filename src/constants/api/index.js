const BASE_URL = 'https://api.exchangeratesapi.io'

export const GET_LATEST_RATE = `${BASE_URL}/latest`