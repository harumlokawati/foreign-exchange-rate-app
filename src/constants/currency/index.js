export const currency = {
    "USD": {
        "symbol": "$",
        "name": "United States Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "USD",
        "name_plural": "US dollars"
    },
    "CAD": {
        "symbol": "CA$",
        "name": "Canadian Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "CAD",
        "name_plural": "Canadian dollars"
    },
    "EUR": {
        "symbol": "€",
        "name": "Euro",
        "symbol_native": "€",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "EUR",
        "name_plural": "euros"
    },
    "AUD": {
        "symbol": "AU$",
        "name": "Australian Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "AUD",
        "name_plural": "Australian dollars"
    },
    "BGN": {
        "symbol": "BGN",
        "name": "Bulgarian Lev",
        "symbol_native": "лв.",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "BGN",
        "name_plural": "Bulgarian leva"
    },
    "BRL": {
        "symbol": "R$",
        "name": "Brazilian Real",
        "symbol_native": "R$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "BRL",
        "name_plural": "Brazilian reals"
    },
    "CHF": {
        "symbol": "CHF",
        "name": "Swiss Franc",
        "symbol_native": "CHF",
        "decimal_digits": 2,
        "rounding": 0.05,
        "code": "CHF",
        "name_plural": "Swiss francs"
    },
    "CNY": {
        "symbol": "CN¥",
        "name": "Chinese Yuan",
        "symbol_native": "CN¥",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "CNY",
        "name_plural": "Chinese yuan"
    },
    "GBP": {
        "symbol": "£",
        "name": "British Pound Sterling",
        "symbol_native": "£",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "GBP",
        "name_plural": "British pounds sterling"
    },
    "HKD": {
        "symbol": "HK$",
        "name": "Hong Kong Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "HKD",
        "name_plural": "Hong Kong dollars"
    },
    "IDR": {
        "symbol": "Rp",
        "name": "Indonesian Rupiah",
        "symbol_native": "Rp",
        "decimal_digits": 0,
        "rounding": 0,
        "code": "IDR",
        "name_plural": "Indonesian rupiahs"
    },
    "INR": {
        "symbol": "Rs",
        "name": "Indian Rupee",
        "symbol_native": "টকা",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "INR",
        "name_plural": "Indian rupees"
    },
    "JPY": {
        "symbol": "¥",
        "name": "Japanese Yen",
        "symbol_native": "￥",
        "decimal_digits": 0,
        "rounding": 0,
        "code": "JPY",
        "name_plural": "Japanese yen"
    },
    "KRW": {
        "symbol": "₩",
        "name": "South Korean Won",
        "symbol_native": "₩",
        "decimal_digits": 0,
        "rounding": 0,
        "code": "KRW",
        "name_plural": "South Korean won"
    },
    "MXN": {
        "symbol": "MX$",
        "name": "Mexican Peso",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "MXN",
        "name_plural": "Mexican pesos"
    },
    "MYR": {
        "symbol": "RM",
        "name": "Malaysian Ringgit",
        "symbol_native": "RM",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "MYR",
        "name_plural": "Malaysian ringgits"
    },
    "NOK": {
        "symbol": "Nkr",
        "name": "Norwegian Krone",
        "symbol_native": "kr",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "NOK",
        "name_plural": "Norwegian kroner"
    },
    "NZD": {
        "symbol": "NZ$",
        "name": "New Zealand Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "NZD",
        "name_plural": "New Zealand dollars"
    },
    "PHP": {
        "symbol": "₱",
        "name": "Philippine Peso",
        "symbol_native": "₱",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "PHP",
        "name_plural": "Philippine pesos"
    },
    "RUB": {
        "symbol": "RUB",
        "name": "Russian Ruble",
        "symbol_native": "руб.",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "RUB",
        "name_plural": "Russian rubles"
    },
    "SDG": {
        "symbol": "SDG",
        "name": "Sudanese Pound",
        "symbol_native": "SDG",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "SDG",
        "name_plural": "Sudanese pounds"
    },
    "SGD": {
        "symbol": "S$",
        "name": "Singapore Dollar",
        "symbol_native": "$",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "SGD",
        "name_plural": "Singapore dollars"
    },
    "THB": {
        "symbol": "฿",
        "name": "Thai Baht",
        "symbol_native": "฿",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "THB",
        "name_plural": "Thai baht"
    },
    "TND": {
        "symbol": "DT",
        "name": "Tunisian Dinar",
        "symbol_native": "د.ت.‏",
        "decimal_digits": 3,
        "rounding": 0,
        "code": "TND",
        "name_plural": "Tunisian dinars"
    },
    "TRY": {
        "symbol": "TL",
        "name": "Turkish Lira",
        "symbol_native": "TL",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "TRY",
        "name_plural": "Turkish Lira"
    },
    "ZAR": {
        "symbol": "R",
        "name": "South African Rand",
        "symbol_native": "R",
        "decimal_digits": 2,
        "rounding": 0,
        "code": "ZAR",
        "name_plural": "South African rand"
    },
}