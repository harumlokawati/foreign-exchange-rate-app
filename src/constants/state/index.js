export const initialState = {
    app:{
            base: 'USD',
            amount: 10,
            symbols: [],
            rates: {},
            error: '',
            loading: false
        }
    }