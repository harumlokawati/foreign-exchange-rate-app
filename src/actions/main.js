import * as actions from '../constants/actions';

export function failure(error) {
    return {
        type: actions.FAILURE,
        payload: {error}
    }
}

export function loadRate(symbols, base) {
    return {
        type: actions.LOAD_RATE,
        payload: {symbols, base}
    }
}

export function loadRateSuccess(data) {
    return {
        type: actions.LOAD_RATE_SUCCESS,
        payload: {data}
    }
}

export function setAmount(amount) {
    return {
        type: actions.SET_AMOUNT,
        payload: {amount}
    }
}

export function setBase(base) {
    return {
        type: actions.SET_BASE,
        payload: {base}
    }
}


export function addSymbol(symbol) {
    return {
        type: actions.ADD_SYMBOL,
        payload: {symbol}
    }
}

export function removeSymbol(symbol) {
    return {
        type: actions.REMOVE_SYMBOL,
        payload: {symbol}
    }
}

export function setLoading(loading) {
    return {
        type: actions.SET_LOADING,
        payload: {loading}
    }
}

