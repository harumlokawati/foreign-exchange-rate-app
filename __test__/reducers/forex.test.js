import forexReducers from '../../src/reducers/forex';
import * as Actions from '../../src/constants/actions';

describe('forex_reducer', () => {
    describe('INITIAL_STATE', () => {
        test('is correct', () => {
            const action = { type: 'dummy_action' };

            expect(forexReducers(undefined, action)).toMatchSnapshot()
        });
    });

    describe('SET_LOADING', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.SET_LOADING, payload: {loading:true }};

            expect(forexReducers(undefined, action)).toMatchSnapshot()
        });
    });

    describe('LOAD_RATE_SUCCESS', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.LOAD_RATE_SUCCESS, payload: {"data":{"base":"EUR","rates":{"USD":1.1187,"GBP":0.88318},"date":"2019-05-24"}} };


            expect(forexReducers(undefined, action)).toMatchSnapshot();
        });
    });

    describe('SET_AMOUNT', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.SET_AMOUNT, payload: {amount:105} };

            expect(forexReducers(undefined, action)).toMatchSnapshot();
        });
    });

    describe('ADD_SYMBOL', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.ADD_SYMBOL, payload: {symbol:'EUR'} };

            expect(forexReducers(undefined, action)).toMatchSnapshot();
        });
    });

    describe('REMOVE_SYMBOL', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.REMOVE_SYMBOL, payload: {symbol:'IDR'} };
            const initial = {
                base: 'USD',
                amount: 105,
                symbols: ['EUR', 'IDR'],
                rates: {},
                error: '',
                loading: false
            }

            expect(forexReducers(initial, action)).toMatchSnapshot();
        });
    });

    describe('SET_BASE', () => {
        test('returns the correct state', () => {
            const action = { type: Actions.SET_BASE, payload: {base:'JPY'} };

            expect(forexReducers(undefined, action)).toMatchSnapshot();
        });
    });
});
