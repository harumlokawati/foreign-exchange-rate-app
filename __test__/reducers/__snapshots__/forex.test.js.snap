// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forex_reducer ADD_SYMBOL returns the correct state 1`] = `
Object {
  "amount": 10,
  "base": "USD",
  "error": "",
  "loading": false,
  "rates": Object {},
  "symbols": Array [
    "EUR",
  ],
}
`;

exports[`forex_reducer INITIAL_STATE is correct 1`] = `
Object {
  "amount": 10,
  "base": "USD",
  "error": "",
  "loading": false,
  "rates": Object {},
  "symbols": Array [],
}
`;

exports[`forex_reducer LOAD_RATE_SUCCESS returns the correct state 1`] = `
Object {
  "amount": 10,
  "base": "USD",
  "error": "",
  "loading": false,
  "rates": Object {
    "GBP": 0.88318,
    "USD": 1.1187,
  },
  "symbols": Array [],
}
`;

exports[`forex_reducer REMOVE_SYMBOL returns the correct state 1`] = `
Object {
  "amount": 105,
  "base": "USD",
  "error": "",
  "loading": false,
  "rates": Object {},
  "symbols": Array [
    "EUR",
  ],
}
`;

exports[`forex_reducer SET_AMOUNT returns the correct state 1`] = `
Object {
  "amount": 105,
  "base": "USD",
  "error": "",
  "loading": false,
  "rates": Object {},
  "symbols": Array [],
}
`;

exports[`forex_reducer SET_BASE returns the correct state 1`] = `
Object {
  "amount": 10,
  "base": "JPY",
  "error": "",
  "loading": false,
  "rates": Object {},
  "symbols": Array [],
}
`;

exports[`forex_reducer SET_LOADING returns the correct state 1`] = `
Object {
  "amount": 10,
  "base": "USD",
  "error": "",
  "loading": true,
  "rates": Object {},
  "symbols": Array [],
}
`;
